apps/meteor/server/services/authorization/canAccessRoom.ts

```diff
@@ -85,6 +105,10 @@ export const canAccessRoom: RoomAccessValidator = async (room, user, extraData):
 	// 	return false;
 	// }
 
+	if (room && user) {
+		if (!(await hasRoomViewPermission(room.t, user._id))) return false;
+	}
+
```

Comment:
```suggestion
```
Do not **block** the access. Instead change each validator to stop **granting** the access when the permission is missing.